Prepare AVAudioSessionCaptureDeviceManager for listing output devices

youennf opened the pull request:
#### 93d31b7aac48f5cb98d7f1c85c8530cd5113c011
<pre>
Prepare AVAudioSessionCaptureDeviceManager for listing output devices
<a href="https://rdar.apple.com/141282717">rdar://141282717</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=284463">https://bugs.webkit.org/show_bug.cgi?id=284463</a>

Reviewed by Eric Carlson.

We do a refactoring of AVAudioSessionCaptureDeviceManager so that it can list output devices as well as microphones.
We soft link some AV strings to identify the type of output device.

* Source/WebCore/PAL/pal/cocoa/AVFoundationSoftLink.h:
* Source/WebCore/PAL/pal/cocoa/AVFoundationSoftLink.mm:
* Source/WebCore/platform/mediastream/ios/AVAudioSessionCaptureDevice.h:
* Source/WebCore/platform/mediastream/ios/AVAudioSessionCaptureDevice.mm:
(WebCore::AVAudioSessionCaptureDevice::createInput):
(WebCore::AVAudioSessionCaptureDevice::createOutput):
(WebCore::AVAudioSessionCaptureDevice::AVAudioSessionCaptureDevice):
(WebCore::AVAudioSessionCaptureDevice::isBuiltin const):
(WebCore::AVAudioSessionCaptureDevice::isLine const):
(WebCore::AVAudioSessionCaptureDevice::isHeadset const):
(WebCore::AVAudioSessionCaptureDevice::isolatedCopy):
(WebCore::AVAudioSessionCaptureDevice::create): Deleted.
* Source/WebCore/platform/mediastream/ios/AVAudioSessionCaptureDeviceManager.h:
* Source/WebCore/platform/mediastream/ios/AVAudioSessionCaptureDeviceManager.mm:
(WebCore::AVAudioSessionCaptureDeviceManager::captureDevices):
(WebCore::AVAudioSessionCaptureDeviceManager::retrieveAudioSessionCaptureDevices const):
(WebCore::AVAudioSessionCaptureDeviceManager::setAudioCaptureDevices):

Canonical link: <a href="https://commits.webkit.org/287758@main">https://commits.webkit.org/287758@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/78be70c80395f1151f7cb898c4fff79e28bd207e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/80704 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/226 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/34640 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/85230 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/31688 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/82815 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/243 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/8023 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/63031 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/20824 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/83773 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/87 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/73459 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/43333 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/24 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/27619 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/30145 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/71585 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/28144 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/86663 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/7931 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/5604 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/71325 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/8108 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/69295 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/70566 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17584 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/14587 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/13532 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/7893 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/13414 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/7732 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/11251 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/9538 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

eric-carlson commented on Source/WebCore/platform/mediastream/ios/AVAudioSessionCaptureDevice.h:
```diff
@@ -36,14 +36,21 @@ namespace WebCore {
     AVAudioSessionCaptureDevice isolatedCopy() &&;
 
+    bool isBuiltin() const;
+    bool isLine() const;
```
Nit: `isLine` is a confusing name. It might be better to have different methods for line-in and line-out, or if it is better to have one method rename to something line `isLineInOrOut`

youennf replied:
Changed to isLineInOrOut

youennf committed:
Prepare AVAudioSessionCaptureDeviceManager for listing output devices
rdar://141282717
https://bugs.webkit.org/show_bug.cgi?id=284463

Reviewed by Eric Carlson.

We do a refactoring of AVAudioSessionCaptureDeviceManager so that it can list output devices as well as microphones.
We soft link some AV strings to identify the type of output device.

* Source/WebCore/PAL/pal/cocoa/AVFoundationSoftLink.h:
* Source/WebCore/PAL/pal/cocoa/AVFoundationSoftLink.mm:
* Source/WebCore/platform/mediastream/ios/AVAudioSessionCaptureDevice.h:
* Source/WebCore/platform/mediastream/ios/AVAudioSessionCaptureDevice.mm:
(WebCore::AVAudioSessionCaptureDevice::createInput):
(WebCore::AVAudioSessionCaptureDevice::createOutput):
(WebCore::AVAudioSessionCaptureDevice::AVAudioSessionCaptureDevice):
(WebCore::AVAudioSessionCaptureDevice::isBuiltin const):
(WebCore::AVAudioSessionCaptureDevice::isLine const):
(WebCore::AVAudioSessionCaptureDevice::isHeadset const):
(WebCore::AVAudioSessionCaptureDevice::isolatedCopy):
(WebCore::AVAudioSessionCaptureDevice::create): Deleted.
* Source/WebCore/platform/mediastream/ios/AVAudioSessionCaptureDeviceManager.h:
* Source/WebCore/platform/mediastream/ios/AVAudioSessionCaptureDeviceManager.mm:
(WebCore::AVAudioSessionCaptureDeviceManager::captureDevices):
(WebCore::AVAudioSessionCaptureDeviceManager::retrieveAudioSessionCaptureDevices const):
(WebCore::AVAudioSessionCaptureDeviceManager::setAudioCaptureDevices):

Canonical link: https://commits.webkit.org/287758@main
